uwblueprint/building-up-client
# Email verification

## Description

Link to ticket: https://www.notion.so/uwblueprintexecs/Add-email-verification-44d2cde7d58c4dbfadd3ce196beae3f7

## Approach

* Added EmailVerify page - the page displayed upon clicking the verification URL provided in the email 
* Added EmailVerificationSent page - the page displayed when a user has not yet been verified
* Added isVerified as a field to the user

## Testing
* Pull the branch email-verification on both building-up-client and building-up-server
* Update your database (you may have to drop and create a new db)
* Register a new account with a valid email address
* You should receive an email in your inbox from hongyichen@uwblueprint.org (temporary email address)
* Clicking the Resend email button will send another email
* If you only copy part of the URL, you should see "Oops..." page
  * Clicking "Return to home" will take you to the EmailVerificationSent page
* If you click on the whole link, you should see "Email Verified" page
  * Clicking "Return to home" will take you to your desired homepage

## Screenshots
<img width="1440" alt="Screen Shot 2021-04-11 at 5 25 08 PM" src="https://user-images.githubusercontent.com/36828558/114321839-f813bd00-9aea-11eb-9c81-fe651c22b08f.png">
<img width="1440" alt="Screen Shot 2021-04-11 at 4 52 35 PM" src="https://user-images.githubusercontent.com/36828558/114321568-6f485180-9ae9-11eb-93ef-c6a4e542662d.png">
<img width="1440" alt="Screen Shot 2021-04-11 at 4 53 32 PM" src="https://user-images.githubusercontent.com/36828558/114321571-70797e80-9ae9-11eb-9b44-73e5d313c9f8.png">
